local_planner/src/nodes/local_planner_node.cpp

```diff
@@ -536,13 +541,30 @@ void LocalPlannerNode::checkPx4Parameters() {
     }
   };
   while (!should_exit_) {
-    request_param("MPC_XY_CRUISE", local_planner_->px4_.param_mpc_xy_cruise);
+    if (!mission_item_speed_set_) {
+      request_param("MPC_XY_CRUISE", local_planner_->px4_.param_mpc_xy_cruise);
+    }
     request_param("MPC_COL_PREV_D", local_planner_->px4_.param_mpc_col_prev_d);
 
     std::this_thread::sleep_for(std::chrono::seconds(30));
   }
 }
 
+void LocalPlannerNode::missionCallback(const mavros_msgs::WaypointList& msg) {
+  for (auto waypoint : msg.waypoints) {
+    if (waypoint.command == 178 && (waypoint.param1 - 1.0f) < FLT_MIN &&
+        waypoint.param2 > 0.0f) {
+      // MAV_CMD_DO_CHANGE_SPEED, speed type: ground speed, speed valid
+      local_planner_->px4_.param_mpc_xy_cruise = waypoint.param2;
```

Comment:
I'm not super happy with setting a named parameter with the value of a different parameter - it might lead to confusion later on. Maybe in our `px4_` we can add a `mission_custom_cruise` that is NaN when not set. Then when we set the speed we check first the custom cruise, and afterwards if it is NaN we take the `mpc_xy_cruise` as a fallback.